nextcord/automod.py

```diff
@@ -0,0 +1,333 @@
+        timeout_seconds: Optional[:class:`int`]
+            The seconds to timeout the person triggered this rule.
+        enabled: Optional[:class:`bool`]
+            Whether if this rule is enabled.
+        exempt_roles: Optional[List[:class:`Role`]]
+            A list of roles that should not be affected by this rule.
+        exempt_channels: Optional[List[:class:`abc.GuildChannel`]]
+            A list of channels that should not be affected by this rule.
+        """
+        payload = {}
+        if "name" in fields:
+            payload["name"] = fields["name"]
+
+        if "event_type" in fields:
+            payload["event_type"] = fields["event_type"].value
+
+        if "keyword_filters" in fields and self.trigger_type != TriggerType.keyword:
+            raise InvalidArgument(
+                "trigger_type must be TriggerType.keyword to pass keyword_filters"
+            )
+
+        if "keyword_filters" in fields:
+            payload["trigger_metadata"]["keyword_filters"] = fields["keyword_filters"]
+
+        if "presets" in fields and self.trigger_type != TriggerType.keyword_preset:
+            raise InvalidArgument("trigger_type must be TriggerType.keyword_preset to pass presets")
+
+        if "presets" in fields:
+            payload["trigger_metadata"]["presets"] = fields["presets"]
+
+        # TODO: notify_channel + timeout_seconds
+
+        if "notify_channel" in fields or "timeout_seconds" in fields:
+            payload["actions"] = []
+
+        if "notify_channel" in fields and not "timeout_seconds" in fields:
+            payload["actions"].append(
+                {"type": 1, "notify_channel_id": fields.get("notify_channel").id}  # type: ignore
+            )
+
+        if "timeout_seconds" in fields and not "notify_channel" in fields:
+            payload["actions"].append({"type": 2, "timeout_seconds": fields.get("timeout_seconds")})
+
+        if "timeout_seconds" in fields and "notify_channel" in fields:
+            payload["actions"].append(
+                {"type": 1, "notify_channel_id": fields.get("notify_channel").id}  # type: ignore
+            )
+            payload["actions"].append(
+                {"type": 2, "timeout_seconds": fields.get("timeout_seconds").id}  # type: ignore
+            )
+
+        if "enabled" in fields:
+            payload["enabled"] = fields["enabled"]
+
+        if "exempt_roles" in fields:
+            payload["exempt_roles"] = [role.id for role in fields["exempt_roles"]]
+
+        if "exempt_channels" in fields:
+            payload["exempt_channels"] = [
+                exempt_channel.id for exempt_channel in fields["exempt_channels"]
+            ]
+
+        if "reason" in fields:
+            payload["reason"] = fields["reason"]
+
+        new_data = await self._state.http.modify_automod_rule(
+            guild_id=self.guild.id, rule_id=self.id, **payload
+        )
+
+        rule = self._state.add_automod_rule(data=new_data)
+        return rule
```

Comment:
```suggestion
        if name is not MISSING:
            payload["name"] = name

        if event_type is not MISSING:
            payload["event_type"] = event_type.value

        if keyword_filters is not MISSING and self.trigger_type != TriggerType.keyword:
            raise InvalidArgument(
                "trigger_type must be TriggerType.keyword to pass keyword_filters"
            )

        if keyword_filters is not MISSING:
            payload["trigger_metadata"]["keyword_filters"] = keyword_filters

        if presets is not MISSING and self.trigger_type != TriggerType.keyword_preset:
            raise InvalidArgument("trigger_type must be TriggerType.keyword_preset to pass presets")

        if presets is not MISSING:
            payload["trigger_metadata"]["presets"] = presets

        # TODO: notify_channel + timeout_seconds

        if notify_channel is not MISSING or timeout_seconds is not MISSING:
            payload["actions"] = []

        if notify_channel is not MISSING and timeout_seconds is MISSING:
            payload["actions"].append(
                {"type": 1, "notify_channel_id": notify_channel.id}
            )

        if timeout_seconds is not MISSING and notify_channel is MISSING:
            payload["actions"].append({"type": 2, "timeout_seconds": timeout_seconds})

        if timeout_seconds is not MISSING and notify_channel is not MISSING:
            payload["actions"].append(
                {"type": 1, "notify_channel_id": notify_channel.id}
            )
            payload["actions"].append(
                {"type": 2, "timeout_seconds": timeout_seconds.id}
            )

        if enabled is not MISSING:
            payload["enabled"] = enabled

        if exempt_roles is not MISSING:
            payload["exempt_roles"] = [role.id for role in exempt_roles]

        if exempt_channels is not MISSING:
            payload["exempt_channels"] = [
                exempt_channel.id for exempt_channel in exempt_channels
            ]

        if reason is not MISSING:
            payload["reason"] = reason

        new_data = await self._state.http.modify_automod_rule(
            guild_id=self.guild.id, rule_id=self.id, **payload
        )

        rule = self._state.add_automod_rule(data=new_data)

        return rule
```
upon applying the suggestion above :)
**DISCLAIMER:** this was all done in github so i have no type checker available for silly mistakes, check this after committed